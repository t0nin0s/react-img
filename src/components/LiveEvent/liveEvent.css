.liveEvent {
  padding: 1.5rem;
  margin: 1.5rem;
}

.liveEvent:not(:last-child) {
  border-bottom: 1px solid black;
}

.info {
  font-size: 1.2rem;
  padding-bottom: 1rem;
}

.tournamentName{
  font-size: 1.5rem;
  font-weight: bold;
}

.draw {
  display: flex;
  justify-content: space-between;
  margin: 0 auto;
}

.vs {
  font-weight: bold;
  font-size: 1.5rem;
}
