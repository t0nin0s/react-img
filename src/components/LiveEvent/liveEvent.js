import React, { Component } from 'react'

import Player from '../Player/Player'

import styles from './liveEvent.css'

class LiveEvent extends Component {
  toLocalTime (time) {
    const separator = time.indexOf('+') > 0 ? '+' : '-'
    return (
      time.split(separator).reduce((orig, local) => {
        if (separator === '+') {
          const localHour = parseInt(orig.split(':')[0]) - parseInt(local.split(':')[0])
          return localHour.toString() + ':00'
        } else {
          const localHour = parseInt(orig.split(':')[0]) + parseInt(local.split(':')[0])
          return localHour.toString() + ':00'
        }
      })
    )
  }

  render () {
    return (
      <div className={styles.liveEvent}>
        <div className={styles.info}>Now Live:
          <span className={styles.tournamentName}> {this.props.tournamentName}</span>
          &nbsp;Started at:&nbsp;
          <span className={styles.time}>{this.props.time} ({this.toLocalTime(this.props.time)} UK Time)</span>
        </div>
        <div className={styles.draw}>
          <Player {...this.props.p1} />
          <span className={styles.vs}>VS</span>
          <Player {...this.props.p2} />
        </div>
      </div>
    )
  }
}

export default LiveEvent
