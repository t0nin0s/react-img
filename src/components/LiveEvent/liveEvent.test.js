import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LiveEvent from './liveEvent'

const liveEvent = {
  id: "2016-11-MS02",
  time: "11:00+01:00",
  p1: {
    id: 2,
    firstName: "Mark",
    lastName: "Steven",
    country: "UK"
  },
  p2: {
    id: 3,
    firstName: "Marcos",
    lastName: "Steven",
    country: "BRZ"
  },
  tournamentName: "UK Cup"
}

test('LiveEvent component renders as expected', () => {
  const component = shallow(<LiveEvent  {...liveEvent} />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
