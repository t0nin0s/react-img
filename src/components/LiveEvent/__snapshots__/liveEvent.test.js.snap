exports[`test LiveEvent component renders as expected 1`] = `
<div
  className="liveEvent">
  <div
    className="info">
    Now Live:
    <span
      className="tournamentName">
       
      UK Cup
    </span>
     Started at: 
    <span
      className="time">
      11:00+01:00
       (
      10:00
       UK Time)
    </span>
  </div>
  <div
    className="draw">
    <Player
      country="UK"
      firstName="Mark"
      id={2}
      lastName="Steven" />
    <span
      className="vs">
      VS
    </span>
    <Player
      country="BRZ"
      firstName="Marcos"
      id={3}
      lastName="Steven" />
  </div>
</div>
`;
